lib/cqm/converter/hds_record.rb

```diff
@@ -163,8 +158,33 @@ def generate_qdm_data_element(dc_fixed_keys, dc_type)
           QDM::FacilityLocation.new.from_json(facility.to_json)
         end
       end
-
+      populate_codesystem_oid(data_element)
+      # iterate over all of the fields
+      # if array or one of the nested types (facility location) dive deeper
+      # if object contains codeSystem and codeSystemOid doesn't have a value then set codeSystemOid to the correct OID
       data_element
     end
+
+    def populate_codesystem_oid(entry)
+      if entry.nil? || entry.is_a?(String) || entry.is_a?(BSON::ObjectId) ||
+         entry.is_a?(Time) || entry.is_a?(Date) || entry.is_a?(Boolean) || entry.is_a?(Integer) || entry.is_a?(Float)
+      elsif entry.is_a?(Array)
+        entry.each { |elem| populate_codesystem_oid(elem) }
+      elsif entry.is_a?(Hash)
+        if entry['codeSystem'] || entry[:codeSystem]
+          name_oid_hash = JSON.parse(File.read(File.join(File.dirname(__FILE__), 'name_oid_map.json')))
+          if name_oid_hash[entry['codeSystem']].nil? && name_oid_hash[entry[:codeSystem]].nil?
+            puts 'ERROR: Could Not Resolve OID For Code System ' + entry['codeSystem']
```

Comment:
I see this error popup when running the converter for the following codeSystems: 
- "ICD-9-PCS" 
- "rxnorm"
- "ICD 10"
- "MyCode"
- "mycode"

Now, clearly atleast 2 of these are not real code systems. But we need to discuss what happens if they aren't converted over / what accounts they are on.

Reply:
Addes support for "ICD-9-PCS", "rxnorm", "ICD 10" names. Will have to discuss what to do about "myCode"